pkg/apis/machine/v1beta1/machine_webhook_test.go

```diff
@@ -229,3 +230,735 @@ func TestDefaultAWSProviderSpec(t *testing.T) {
+						StorageAccountType: "storageAccountType",
+					},
+				},
+			},
+			expectedOk:    false,
+			expectedError: "providerSpec.osDisk.osType: Required value: osType must be provided",
+		},
+		{
+			testCase: "with no os disk storage account type it fails",
+			providerSpec: &azure.AzureMachineProviderSpec{
+				Location: "location",
+				VMSize:   "vmSize",
+				Image: azure.Image{
+					ResourceID: "resourceID",
+				},
+				ManagedIdentity: "managedIdentity",
+				ResourceGroup:   "resourceGroup",
+				UserDataSecret: &corev1.SecretReference{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				CredentialsSecret: &corev1.SecretReference{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				OSDisk: azure.OSDisk{
+					DiskSizeGB: 1,
+					OSType:     "osType",
+				},
+			},
+			expectedOk:    false,
+			expectedError: "providerSpec.osDisk.managedDisk.storageAccountType: Required value: storageAccountType must be provided",
+		},
+		{
+			testCase: "with all required fields it succeeds",
+			providerSpec: &azure.AzureMachineProviderSpec{
+				Location: "location",
+				VMSize:   "vmSize",
+				Image: azure.Image{
+					ResourceID: "resourceID",
+				},
+				ManagedIdentity: "managedIdentity",
+				ResourceGroup:   "resourceGroup",
+				UserDataSecret: &corev1.SecretReference{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				CredentialsSecret: &corev1.SecretReference{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				OSDisk: azure.OSDisk{
+					DiskSizeGB: 1,
+					OSType:     "osType",
+					ManagedDisk: azure.ManagedDisk{
+						StorageAccountType: "storageAccountType",
+					},
+				},
+			},
+			expectedOk:    true,
+			expectedError: "",
+		},
+	}
+
+	h := createMachineValidator(osconfigv1.AzurePlatformType, "clusterID")
+
+	for _, tc := range testCases {
+		t.Run(tc.testCase, func(t *testing.T) {
+			m := &Machine{}
+			rawBytes, err := json.Marshal(tc.providerSpec)
+			if err != nil {
+				t.Fatal(err)
+			}
+			m.Spec.ProviderSpec.Value = &runtime.RawExtension{Raw: rawBytes}
+
+			ok, err := h.webhookOperations(h, m)
+			if ok != tc.expectedOk {
+				t.Errorf("expected: %v, got: %v", tc.expectedOk, ok)
+			}
+
+			if err == nil {
+				if tc.expectedError != "" {
+					t.Errorf("expected: %q, got: %v", tc.expectedError, err)
+				}
+			} else {
+				if err.Error() != tc.expectedError {
+					t.Errorf("expected: %q, got: %q", tc.expectedError, err.Error())
+				}
+			}
+		})
+	}
+}
+
+func TestDefaultAzureProviderSpec(t *testing.T) {
+
+	clusterID := "clusterID"
+	testCases := []struct {
+		testCase             string
+		providerSpec         *azure.AzureMachineProviderSpec
+		expectedProviderSpec *azure.AzureMachineProviderSpec
+		expectedError        string
+		expectedOk           bool
+	}{
+		{
+			testCase:     "it defaults defaultable fields",
+			providerSpec: &azure.AzureMachineProviderSpec{},
+			expectedProviderSpec: &azure.AzureMachineProviderSpec{
```

Comment:
Most of test cases are using valid provider spec, but then altering a single field. Could we just patch full spec with changes we need to test, to improve readability?